### PR TITLE
Camera tracking modes selection with GPS and North as options

### PR DESCRIPTION
**Pull request by LukasPaczos**
Added `NAVIGATION_TRACKING_MODE_GPS` and `NAVIGATION_TRACKING_MODE_NORTH` as camera tracking options that can be set via `NavigationMapboxMap#updateCameraTrackingMode`.

ezgif com-video-to-gif 5

Code used for the example:
```
    navigationView.retrieveNavigationMapboxMap().retrieveMap().addOnMapClickListener(new MapboxMap.OnMapClickListener() {

      private boolean isNorth;
      private Camera standardEngine = navigationView.retrieveMapboxNavigation().getCameraEngine();
      private Camera northEngine = new NorthCamera();

      @Override
      public void onMapClick(@NonNull LatLng point) {
        navigationView.retrieveNavigationMapboxMap().updateCameraTrackingMode(
          isNorth ? NavigationCamera.NAVIGATION_TRACKING_MODE_GPS : NavigationCamera.NAVIGATION_TRACKING_MODE_NORTH
        );
        navigationView.retrieveMapboxNavigation().setCameraEngine(
          isNorth ? standardEngine : northEngine
        );
        isNorth = !isNorth;
      }
    });
```
and
```
  private static class NorthCamera extends SimpleCamera {
    @Override
    public double tilt(RouteInformation routeInformation) {
      return 0;
    }
  }
```

### CODE REVIEW

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/camera/SimpleCamera.java [hunk 1]
```diff
@@ -63,15 +50,11 @@ private void buildRouteCoordinatesFromRouteData(RouteInformation routeInformatio
   }
 
   private void setupLineStringAndBearing(DirectionsRoute route) {
-    if (initialRoute != null && route.equals(initialRoute)) {
+    if (route.equals(initialRoute)) {
       return; //no need to recalculate these values
     }
     initialRoute = route;
     routeCoordinates = generateRouteCoordinates(route);
```
Comment (danesfeder) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/camera/SimpleCamera.java]: @LukasPaczos can we double check that these `routeCoordinates` are still being used?  I believe they are (for the route overview `LatLngBounds`) but just want to double check - thanks!

**Comment by hurrba** [comment]
it's nice to see that you finally have started on implementing this! however, there is a problem with the proposed solution: the puck is still in the bottom of the screen, which means that you have very poor overview when driving south. The puck should be in the middle of the screen when `NAVIGATION_TRACKING_MODE_NORTH` is enabled. 

The only way we managed to solve this way to remove the `wayname` label. For some reason, this label prevents setting the puck in the middle of the screen. I haven't had time to look more into it in order to make a pull request, but perhaps it's an easy fix for you?

**Comment by danesfeder** [comment]
@hurrba are you adjusting with the `MapboxMap#setPadding`?  We use the top padding value here to "push" the puck to the bottom above the BottomSheet.  

You should be able to set the tracking mode with this PR and at the same time use `NavigationMapboxMap#retrieveMap#setPadding` to updated the padding you desire (you can cache / then reset to our calculated padding with `getPadding` prior to doing this).

We could do this for you when you set the tracking mode, but that would couple the padding adjustment and other developers may not desire this same behavior.

**Comment by hurrba** [comment]
@danesfeder that seems to do the trick! 
then my only comment is to perhaps add this to the documentation now that it's possible to set `NAVIGATION_TRACKING_MODE_NORTH`. :)

**Comment by Guardiola31337** [comment]
@hurrba 
> then my only comment is to perhaps add this to the documentation now that it's possible to set `NAVIGATION_TRACKING_MODE_NORTH`. :)

Will be for sure! Thanks for the suggestion and for checking out the SDK!

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/camera/SimpleCamera.java (hunk 1 above)
Comment (LukasPaczos) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/camera/SimpleCamera.java]: Looks like they are being used for the overview as you mentioned :)

**Comment by hurrba** [comment]
@LukasPaczos @danesfeder @Guardiola31337  this is still an issue when showing wayname. see https://github.com/mapbox/mapbox-navigation-android/issues/1465

**Pull request merged**